lib/compress/zstd_opt.c

```diff
@@ -454,6 +465,59 @@ repcodes_t ZSTD_updateRep(U32 const rep[3], U32 const offset, U32 const ll0)
         memcpy(opt[pos].rep, &rep_, sizeof(rep_));            \
     }
 
+typedef struct {
+    const BYTE* anchor;
+    U32 litlen;
+    U32 rawLitCost;
+} cachedLiteralPrice_t;
+
+static U32 ZSTD_rawLiteralsCost_cached(
```

Comment:
Does caching this actually make a difference? I saw that in silesia it triggers < 1% of the time in the dev branch.

Reply:
As discussed, it likely matters for some corner cases, when `litlen` becomes very big, requiring a match found after a long section without any match. In which case, the literal section will be scanned several times (at least 3 times, more likely 4-5 times).
In most situations though, I expect the size of a literals section to remain small, so the speed difference would not be visible.

Observing the difference will likely require a data source that can produce such corner case.